driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -114,6 +194,26 @@ public void onResult(final List<RawBsonDocument> result, final Throwable t) {
     }
 
     private void retryOperation(final AsyncBlock asyncBlock, final SingleResultCallback<List<RawBsonDocument>> callback) {
+        changeStreamOperation.startAfter(null);
+        if (resumeToken != null) {
+            changeStreamOperation.startAtOperationTime(null);
+            changeStreamOperation.resumeAfter(resumeToken);
+        } else if (changeStreamOperation.getStartAtOperationTime() != null) {
+            getMaxWireVersion(new SingleResultCallback<Integer>() {
+                @Override
+                public void onResult(final Integer result, final Throwable t) {
+                    if (result != null && result >= 7) {
```

Comment:
I would say, we do not need to check maxWireVersion when we make restoring operation. For me, it's enough if we fill  `startAtOperationTime` from initialAggregate response only for `MaxWireVersion` > 7 (and some other required conditions). At least it's how we implemented it in the c# driver.

In this case (and taking into account my comment in `AsyncChangeStreamBatchCursor` constructor) we do not need `getMaxWireVersion` method in `AsyncChangeStreamBatchCursor` at all.

Reply:
Thoughts?

Reply:
I have removed the check in the implementation of the new methods in ChangeStreamOperation.